integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids.tests.common
+
+import java.io.{File, FileOutputStream}
+import java.time.Instant
+import java.util.concurrent.TimeUnit.NANOSECONDS
+
+import scala.collection.mutable.ListBuffer
+
+import org.json4s.DefaultFormats
+import org.json4s.jackson.JsonMethods.parse
+import org.json4s.jackson.Serialization.writePretty
+
+import org.apache.spark.{SPARK_BUILD_USER, SPARK_VERSION}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+
+object BenchUtils {
+
+  /**
+   * Run the specified number of cold and hot runs and record the timings and summary of the
+   * query and results to file, including all Spark configuration options and environment
+   * variables.
+   */
+  def runBench(
+      spark: SparkSession,
+      queryRunner: SparkSession => DataFrame,
+      queryDescription: String,
+      filenameStub: String,
+      numColdRuns: Int = 1,
+      numHotRuns: Int = 3,
+      maxResultsToRecord: Int = 100): Unit = {
+
+    val now = Instant.now()
+
+    var df: DataFrame = null
+    var results: Seq[Row] = null
+    val coldRunElapsed = new ListBuffer[Long]()
+    for (i <- 0 until numColdRuns) {
+      println(s"*** Start cold run $i:")
+      val start = System.nanoTime()
+      df = queryRunner(spark)
+      results = df.collect
```

Comment:
So I think for smaller data sizes `.collect` will be fine. But I think it may be better to write using one of the writers (as an option). In fact, for benchmark runs I would rather write to disk as parquet, orc, or something like that, for both the cpu and the gpu, this would both include the writer as part of the timings and you could argue, it makes a more realistic test.

Reply:
Sure, this code is just a refactor of how we were already running these benchmarks, but I agree that adding an option to write the results out would be a great follow up.